### PR TITLE
Add Bookmarks Tab to Home Page

### LINKED ISSUE #198: New tab: bookmarks list
simple list with swipe to delete

For the reference check Cliqz 2.0 from App Store.

### PR DESCRIPTION
**Pull request by winsmith**
## Ticket
Fixes #198 

## Description
Includes a new LibaryView for Bookmarks in the Home View

## Checklist:

- [x] My squashed commit messages follows the [7 golden rules](https://chris.beams.io/posts/git-commit/)
- [x] My code follows the code style of this project.
- [x] I updated or created necessary unit tests
- [x] I updated all necessary documentation

### CODE REVIEW

File: UserAgent/Home View/BookmarksView.swift [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    fileprivate func onModelFailure(_ e: Any) {
+        print("Error: failed to get data: \(e)")
+    }
+
+}
+
+// MARK: - Private API
+private extension BookmarksView {
+    private func updateEmptyPanelState() {
+        // TOOD
```
Comment (shahenAnt) [inline UserAgent/Home View/BookmarksView.swift]: As we decided to create new ticket for Todos. Please create new ticket for this todo.

File: UserAgent/Home View/BookmarksView.swift [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        static let IconBorderColor = UIColor.Grey30
+    }
+
+    // MARK: - Properties
+    var source: BookmarksModel?
+    var parentFolders = [BookmarkFolder]()
+    var bookmarkFolder: BookmarkFolder?
+    var refreshControl: UIRefreshControl?
+
+    fileprivate lazy var emptyStateOverlayView: UIView = self.createEmptyStateOverlayView()
```
Comment (shahenAnt) [inline UserAgent/Home View/BookmarksView.swift]: Do we need to use fileprivate or private is enough?

File: UserAgent/Home View/BookmarksView.swift [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+    override func pinToTopSites(_ site: Site) {
+        _ = self.profile.history.addPinnedTopSite(site).value
+    }
+
+    override func reloadData() {
+        self.source?.reloadData().upon(onModelFetched)
+    }
+
+    func loadData() {
```
Comment (shahenAnt) [inline UserAgent/Home View/BookmarksView.swift]: I think loadData() method should be private?

File: UserAgent/Home View/BookmarksView.swift [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+
+import Shared
+import Storage
+import UIKit
+
+// MARK: - Placeholder strings for Bug 1232810.
+let deleteWarningTitle = NSLocalizedString("This folder isn’t empty.", tableName: "BookmarkPanelDeleteConfirm", comment: "Title of the confirmation alert when the user tries to delete a folder that still contains bookmarks and/or folders.")
```
Comment (shahenAnt) [inline UserAgent/Home View/BookmarksView.swift]: Do we need to move all localizations in String extension under Shared folder? Let's discuss it during today's meeting.

File: UserAgent/Home View/BookmarksView.swift (hunk 1 above)
Comment (winsmith) [reply UserAgent/Home View/BookmarksView.swift]: Whoops you caught me. It wasn't highlighted because of the typo. THanks

**Comment by chrmod** [comment]
I've checked the failing tests and those are not related to this PR. It is fine to merge.

**Pull request merged**